### ISSUE TITLE
memory usage problem

### ISSUE THREAD

**Issue by bkujszczyk** [sabeechen/hassio-google-drive-backup#1085]
Ha and proxmox
During the backup, the memory usage increases from 2.9 GB to almost 6 GB. After the process is finished, the memory is not released. You have to restart the entire machine. And since Zigbee2MQTT does not like to have few resources, after each backup some devices switch to off-network mode

**Comment by glennwalsh** [comment]
I also am experiencing the same issue @bkujszczyk . As part of my testing i stopped the addon, created a back via HA, the supervisor memory shot up as expected and then dropped when backup was complete. I then started Google Backup addon, backup did sync, the memory usage shot up as expected but then dropped.

**Comment by strpc** [comment]
hi! same problem.

HA OS 2025.11 in Proxmox(vm)

<img width="1448" height="760" alt="Image" src="https://github.com/user-attachments/assets/75c5c2e0-c388-4b8a-96e9-a40f7c5847a0" />

<details><summary>environment details</summary>
<pre>
Core version-2025.11.2
Installation type of Home Assistant OS
The development environment is false
Supervisor	true
Docker	true
Amd64 Container Architecture
The root user
The virtual environment is false
Python version 3.13.9
The Linux family of operating systems
Operating system version 6.12.51-haos
x86_64 CPU Architecture
Europe/Moscow time zone
Configuration directory /config
</pre>
</details> 

<details><summary>log</summary>
<pre>
11-21 14:07:16 DEBUG [backup.model.syncer] Sync requested by Coordinator
11-21 14:07:16 INFO [backup.model.coordinator] Syncing Backups
11-21 14:07:16 INFO [backup.ha.hasource] Requesting a new backup
11-21 14:07:16 INFO [backup.watcher] A backup directory file was modified, we'll check for new backups soon.
11-21 14:07:26 DEBUG [backup.watcher] Checking backup source for changes...
11-21 14:07:41 INFO [backup.watcher] A backup directory file was modified, we'll check for new backups soon.
11-21 14:07:41 INFO [backup.ha.hasource] Backup finished
11-21 14:07:41 DEBUG [backup.model.syncer] Sync requested by HomeAssistant
11-21 14:07:41 INFO [backup.model.coordinator] Syncing Backups
11-21 14:07:42 INFO [backup.ha.hasource] Deleting 'Full Backup 2025-11-17 14:07:15' from Home Assistant
11-21 14:07:42 INFO [backup.drive.drivesource] Uploading 'Full Backup 2025-11-21 14:07:16' to Google Drive
11-21 14:07:42 DEBUG [backup.drive.driverequests] Starting a new upload session with Google Drive
11-21 14:07:43 DEBUG [backup.drive.driverequests] Sending 256 KB to Google Drive
11-21 14:07:44 DEBUG [backup.drive.drivesource] Uploading Full Backup 2025-11-21 14:07:16 0.04%
11-21 14:07:44 DEBUG [backup.drive.driverequests] Sending 4.25 MB to Google Drive
11-21 14:07:44 DEBUG [backup.drive.drivesource] Uploading Full Backup 2025-11-21 14:07:16 0.64%
11-21 14:07:45 DEBUG [backup.drive.driverequests] Sending 10 MB to Google Drive
11-21 14:07:46 DEBUG [backup.drive.drivesource] Uploading Full Backup 2025-11-21 14:07:16 2.05%
11-21 14:07:46 DEBUG [backup.drive.driverequests] Sending 10 MB to Google Drive
11-21 14:07:47 DEBUG [backup.drive.drivesource] Uploading Full Backup 2025-11-21 14:07:16 3.47%
11-21 14:07:47 DEBUG [backup.drive.driverequests] Sending 10 MB to Google Drive
11-21 14:07:49 DEBUG [backup.drive.drivesource] Uploading Full Backup 2025-11-21 14:07:16 4.88%
11-21 14:07:49 DEBUG [backup.drive.driverequests] Sending 10 MB to Google Drive
11-21 14:07:50 DEBUG [backup.drive.drivesource] Uploading Full Backup 2025-11-21 14:07:16 6.30%
11-21 14:07:50 DEBUG [backup.drive.driverequests] Sending 10 MB to Google Drive
11-21 14:07:51 DEBUG [backup.drive.drivesource] Uploading Full Backup 2025-11-21 14:07:16 7.71%
11-21 14:07:51 DEBUG [backup.drive.driverequests] Sending 10 MB to Google Drive
11-21 14:07:52 DEBUG [backup.drive.drivesource] Uploading Full Backup 2025-11-21 14:07:16 9.13%
11-21 14:07:52 DEBUG [backup.drive.driverequests] Sending 10 MB to Google Drive
11-21 14:07:53 DEBUG [backup.drive.drivesource] Uploading Full Backup 2025-11-21 14:07:16 10.54%
11-21 14:07:53 DEBUG [backup.drive.driverequests] Sending 10 MB to Google Drive
11-21 14:07:55 DEBUG [backup.drive.drivesource] Uploading Full Backup 2025-11-21 14:07:16 11.96%
11-21 14:07:55 DEBUG [backup.drive.driverequests] Sending 10 MB to Google Drive
11-21 14:07:57 DEBUG [backup.drive.drivesource] Uploading Full Backup 2025-11-21 14:07:16 13.37%
11-21 14:07:57 DEBUG [backup.drive.driverequests] Sending 10 MB to Google Drive
11-21 14:07:58 DEBUG [backup.drive.drivesource] Uploading Full Backup 2025-11-21 14:07:16 14.79%
11-21 14:07:58 DEBUG [backup.drive.driverequests] Sending 10 MB to Google Drive
11-21 14:08:00 DEBUG [backup.drive.drivesource] Uploading Full Backup 2025-11-21 14:07:16 16.20%
11-21 14:08:00 DEBUG [backup.drive.driverequests] Sending 10 MB to Google Drive
11-21 14:08:02 DEBUG [backup.drive.drivesource] Uploading Full Backup 2025-11-21 14:07:16 17.62%
11-21 14:08:02 DEBUG [backup.drive.driverequests] Sending 10 MB to Google Drive
11-21 14:08:03 DEBUG [backup.drive.drivesource] Uploading Full Backup 2025-11-21 14:07:16 19.03%
11-21 14:08:03 DEBUG [backup.drive.driverequests] Sending 10 MB to Google Drive
11-21 14:08:05 DEBUG [backup.drive.drivesource] Uploading Full Backup 2025-11-21 14:07:16 20.45%
11-21 14:08:05 DEBUG [backup.drive.driverequests] Sending 10 MB to Google Drive
11-21 14:08:07 DEBUG [backup.drive.drivesource] Uploading Full Backup 2025-11-21 14:07:16 21.86%
11-21 14:08:07 DEBUG [backup.drive.driverequests] Sending 10 MB to Google Drive
11-21 14:08:09 DEBUG [backup.drive.drivesource] Uploading Full Backup 2025-11-21 14:07:16 23.28%
11-21 14:08:09 DEBUG [backup.drive.driverequests] Sending 10 MB to Google Drive
11-21 14:08:10 DEBUG [backup.drive.drivesource] Uploading Full Backup 2025-11-21 14:07:16 24.69%
11-21 14:08:10 DEBUG [backup.drive.driverequests] Sending 10 MB to Google Drive
11-21 14:08:12 DEBUG [backup.drive.drivesource] Uploading Full Backup 2025-11-21 14:07:16 26.11%
11-21 14:08:12 DEBUG [backup.drive.driverequests] Sending 10 MB to Google Drive
11-21 14:08:14 DEBUG [backup.drive.drivesource] Uploading Full Backup 2025-11-21 14:07:16 27.52%
11-21 14:08:14 DEBUG [backup.drive.driverequests] Sending 10 MB to Google Drive
11-21 14:08:17 DEBUG [backup.drive.drivesource] Uploading Full Backup 2025-11-21 14:07:16 28.94%
11-21 14:08:17 DEBUG [backup.drive.driverequests] Sending 10 MB to Google Drive
11-21 14:08:20 DEBUG [backup.drive.drivesource] Uploading Full Backup 2025-11-21 14:07:16 30.36%
11-21 14:08:21 DEBUG [backup.drive.driverequests] Sending 10 MB to Google Drive
11-21 14:08:25 DEBUG [backup.drive.drivesource] Uploading Full Backup 2025-11-21 14:07:16 31.77%
11-21 14:08:25 DEBUG [backup.drive.driverequests] Sending 10 MB to Google Drive
11-21 14:08:29 DEBUG [backup.drive.drivesource] Uploading Full Backup 2025-11-21 14:07:16 33.19%
11-21 14:08:29 DEBUG [backup.drive.driverequests] Sending 10 MB to Google Drive
11-21 14:08:32 DEBUG [backup.drive.drivesource] Uploading Full Backup 2025-11-21 14:07:16 34.60%
11-21 14:08:32 DEBUG [backup.drive.driverequests] Sending 10 MB to Google Drive
11-21 14:08:35 DEBUG [backup.drive.drivesource] Uploading Full Backup 2025-11-21 14:07:16 36.02%
11-21 14:08:35 DEBUG [backup.drive.driverequests] Sending 10 MB to Google Drive
11-21 14:08:36 DEBUG [backup.drive.drivesource] Uploading Full Backup 2025-11-21 14:07:16 37.43%
11-21 14:08:36 DEBUG [backup.drive.driverequests] Sending 10 MB to Google Drive
11-21 14:08:38 DEBUG [backup.drive.drivesource] Uploading Full Backup 2025-11-21 14:07:16 38.85%
11-21 14:08:38 DEBUG [backup.drive.driverequests] Sending 10 MB to Google Drive
11-21 14:08:39 DEBUG [backup.drive.drivesource] Uploading Full Backup 2025-11-21 14:07:16 40.26%
11-21 14:08:39 DEBUG [backup.drive.driverequests] Sending 10 MB to Google Drive
11-21 14:08:40 DEBUG [backup.drive.drivesource] Uploading Full Backup 2025-11-21 14:07:16 41.68%
11-21 14:08:40 DEBUG [backup.drive.driverequests] Sending 10 MB to Google Drive
11-21 14:08:42 DEBUG [backup.drive.drivesource] Uploading Full Backup 2025-11-21 14:07:16 43.09%
11-21 14:08:42 DEBUG [backup.drive.driverequests] Sending 10 MB to Google Drive
11-21 14:08:43 DEBUG [backup.drive.drivesource] Uploading Full Backup 2025-11-21 14:07:16 44.51%
11-21 14:08:44 DEBUG [backup.drive.driverequests] Sending 10 MB to Google Drive
11-21 14:08:45 DEBUG [backup.drive.drivesource] Uploading Full Backup 2025-11-21 14:07:16 45.92%
11-21 14:08:45 DEBUG [backup.drive.driverequests] Sending 10 MB to Google Drive
11-21 14:08:48 DEBUG [backup.drive.drivesource] Uploading Full Backup 2025-11-21 14:07:16 47.34%
11-21 14:08:48 DEBUG [backup.drive.driverequests] Sending 10 MB to Google Drive
11-21 14:08:50 DEBUG [backup.drive.drivesource] Uploading Full Backup 2025-11-21 14:07:16 48.75%
11-21 14:08:50 DEBUG [backup.drive.driverequests] Sending 10 MB to Google Drive
11-21 14:08:52 DEBUG [backup.drive.drivesource] Uploading Full Backup 2025-11-21 14:07:16 50.17%
11-21 14:08:52 DEBUG [backup.drive.driverequests] Sending 10 MB to Google Drive
11-21 14:08:54 DEBUG [backup.drive.drivesource] Uploading Full Backup 2025-11-21 14:07:16 51.58%
11-21 14:08:54 DEBUG [backup.drive.driverequests] Sending 10 MB to Google Drive
11-21 14:08:56 DEBUG [backup.drive.drivesource] Uploading Full Backup 2025-11-21 14:07:16 53.00%
11-21 14:08:56 DEBUG [backup.drive.driverequests] Sending 10 MB to Google Drive
11-21 14:08:58 DEBUG [backup.drive.drivesource] Uploading Full Backup 2025-11-21 14:07:16 54.41%
11-21 14:08:58 DEBUG [backup.drive.driverequests] Sending 10 MB to Google Drive
11-21 14:09:00 DEBUG [backup.drive.drivesource] Uploading Full Backup 2025-11-21 14:07:16 55.83%
11-21 14:09:00 DEBUG [backup.drive.driverequests] Sending 10 MB to Google Drive
11-21 14:09:02 DEBUG [backup.drive.drivesource] Uploading Full Backup 2025-11-21 14:07:16 57.24%
11-21 14:09:02 DEBUG [backup.drive.driverequests] Sending 10 MB to Google Drive
11-21 14:09:04 DEBUG [backup.drive.drivesource] Uploading Full Backup 2025-11-21 14:07:16 58.66%
11-21 14:09:04 DEBUG [backup.drive.driverequests] Sending 10 MB to Google Drive
11-21 14:09:06 DEBUG [backup.drive.drivesource] Uploading Full Backup 2025-11-21 14:07:16 60.07%
11-21 14:09:06 DEBUG [backup.drive.driverequests] Sending 10 MB to Google Drive
11-21 14:09:09 DEBUG [backup.drive.drivesource] Uploading Full Backup 2025-11-21 14:07:16 61.49%
11-21 14:09:09 DEBUG [backup.drive.driverequests] Sending 10 MB to Google Drive
11-21 14:09:11 DEBUG [backup.drive.drivesource] Uploading Full Backup 2025-11-21 14:07:16 62.90%
11-21 14:09:11 DEBUG [backup.drive.driverequests] Sending 10 MB to Google Drive
11-21 14:09:14 DEBUG [backup.drive.drivesource] Uploading Full Backup 2025-11-21 14:07:16 64.32%
11-21 14:09:14 DEBUG [backup.drive.driverequests] Sending 10 MB to Google Drive
11-21 14:09:17 DEBUG [backup.drive.drivesource] Uploading Full Backup 2025-11-21 14:07:16 65.73%
11-21 14:09:17 DEBUG [backup.drive.driverequests] Sending 10 MB to Google Drive
11-21 14:09:19 DEBUG [backup.drive.drivesource] Uploading Full Backup 2025-11-21 14:07:16 67.15%
11-21 14:09:19 DEBUG [backup.drive.driverequests] Sending 10 MB to Google Drive
11-21 14:09:22 DEBUG [backup.drive.drivesource] Uploading Full Backup 2025-11-21 14:07:16 68.56%
11-21 14:09:22 DEBUG [backup.drive.driverequests] Sending 10 MB to Google Drive
11-21 14:09:24 DEBUG [backup.drive.drivesource] Uploading Full Backup 2025-11-21 14:07:16 69.98%
11-21 14:09:24 DEBUG [backup.drive.driverequests] Sending 10 MB to Google Drive
11-21 14:09:26 DEBUG [backup.drive.drivesource] Uploading Full Backup 2025-11-21 14:07:16 71.40%
11-21 14:09:26 DEBUG [backup.drive.driverequests] Sending 10 MB to Google Drive
11-21 14:09:28 DEBUG [backup.drive.drivesource] Uploading Full Backup 2025-11-21 14:07:16 72.81%
11-21 14:09:28 DEBUG [backup.drive.driverequests] Sending 10 MB to Google Drive
11-21 14:09:30 DEBUG [backup.drive.drivesource] Uploading Full Backup 2025-11-21 14:07:16 74.23%
11-21 14:09:30 DEBUG [backup.drive.driverequests] Sending 10 MB to Google Drive
11-21 14:09:31 DEBUG [backup.drive.drivesource] Uploading Full Backup 2025-11-21 14:07:16 75.64%
11-21 14:09:31 DEBUG [backup.drive.driverequests] Sending 10 MB to Google Drive
11-21 14:09:33 DEBUG [backup.drive.drivesource] Uploading Full Backup 2025-11-21 14:07:16 77.06%
11-21 14:09:33 DEBUG [backup.drive.driverequests] Sending 10 MB to Google Drive
11-21 14:09:34 DEBUG [backup.drive.drivesource] Uploading Full Backup 2025-11-21 14:07:16 78.47%
11-21 14:09:34 DEBUG [backup.drive.driverequests] Sending 10 MB to Google Drive
11-21 14:09:35 DEBUG [backup.drive.drivesource] Uploading Full Backup 2025-11-21 14:07:16 79.89%
11-21 14:09:35 DEBUG [backup.drive.driverequests] Sending 10 MB to Google Drive
11-21 14:09:37 DEBUG [backup.drive.drivesource] Uploading Full Backup 2025-11-21 14:07:16 81.30%
11-21 14:09:37 DEBUG [backup.drive.driverequests] Sending 10 MB to Google Drive
11-21 14:09:38 DEBUG [backup.drive.drivesource] Uploading Full Backup 2025-11-21 14:07:16 82.72%
11-21 14:09:38 DEBUG [backup.drive.driverequests] Sending 10 MB to Google Drive
11-21 14:09:39 DEBUG [backup.drive.drivesource] Uploading Full Backup 2025-11-21 14:07:16 84.13%
11-21 14:09:39 DEBUG [backup.drive.driverequests] Sending 10 MB to Google Drive
11-21 14:09:41 DEBUG [backup.drive.drivesource] Uploading Full Backup 2025-11-21 14:07:16 85.55%
11-21 14:09:41 DEBUG [backup.drive.driverequests] Sending 10 MB to Google Drive
11-21 14:09:42 DEBUG [backup.drive.drivesource] Uploading Full Backup 2025-11-21 14:07:16 86.96%
11-21 14:09:42 DEBUG [backup.drive.driverequests] Sending 10 MB to Google Drive
11-21 14:09:43 DEBUG [backup.drive.drivesource] Uploading Full Backup 2025-11-21 14:07:16 88.38%
11-21 14:09:43 DEBUG [backup.drive.driverequests] Sending 10 MB to Google Drive
11-21 14:09:45 DEBUG [backup.drive.drivesource] Uploading Full Backup 2025-11-21 14:07:16 89.79%
11-21 14:09:45 DEBUG [backup.drive.driverequests] Sending 10 MB to Google Drive
11-21 14:09:46 DEBUG [backup.drive.drivesource] Uploading Full Backup 2025-11-21 14:07:16 91.21%
11-21 14:09:46 DEBUG [backup.drive.driverequests] Sending 10 MB to Google Drive
11-21 14:09:48 DEBUG [backup.drive.drivesource] Uploading Full Backup 2025-11-21 14:07:16 92.62%
11-21 14:09:48 DEBUG [backup.drive.driverequests] Sending 10 MB to Google Drive
11-21 14:09:49 DEBUG [backup.drive.drivesource] Uploading Full Backup 2025-11-21 14:07:16 94.04%
11-21 14:09:49 DEBUG [backup.drive.driverequests] Sending 10 MB to Google Drive
11-21 14:09:51 DEBUG [backup.drive.drivesource] Uploading Full Backup 2025-11-21 14:07:16 95.45%
11-21 14:09:51 DEBUG [backup.drive.driverequests] Sending 10 MB to Google Drive
11-21 14:09:53 DEBUG [backup.drive.drivesource] Uploading Full Backup 2025-11-21 14:07:16 96.87%
11-21 14:09:53 DEBUG [backup.drive.driverequests] Sending 10 MB to Google Drive
11-21 14:09:55 DEBUG [backup.drive.drivesource] Uploading Full Backup 2025-11-21 14:07:16 98.28%
11-21 14:09:55 DEBUG [backup.drive.driverequests] Sending 10 MB to Google Drive
11-21 14:09:58 DEBUG [backup.drive.drivesource] Uploading Full Backup 2025-11-21 14:07:16 99.70%
11-21 14:09:58 DEBUG [backup.drive.driverequests] Sending 2.130859375 MB to Google Drive
11-21 14:10:01 DEBUG [backup.drive.drivesource] Uploading Full Backup 2025-11-21 14:07:16 100.00%
11-21 14:10:01 INFO [backup.drive.drivesource] Deleting 'Full Backup 2025-11-06 14:07:10' From Google Drive
11-21 14:10:02 DEBUG [backup.watcher] Checking backup source for changes...
11-21 14:10:02 DEBUG [backup.model.syncer] Sync requested by Backup Directory Watcher
11-21 14:10:02 INFO [backup.model.coordinator] Syncing Backups
</pre>
</details> 


@bkujszczyk have you fixed this leak?
i need help. any details - just tell me.


**update after 30 minutes:**
```
               total        used        free      shared  buff/cache   available
Mem:           5.8Gi       1.1Gi       4.3Gi       5.5Mi       2.4Gi       4.7Gi
Swap:          1.9Gi          0B       1.9Gi
```

after apply: 
```
sync
echo 3 | sudo tee /proc/sys/vm/drop_caches
```

result:
```
free -h
               total        used        free      shared  buff/cache   available
Mem:           5.8Gi       1.1Gi       4.3Gi       5.5Mi       485Mi       4.7Gi
Swap:          1.9Gi          0B       1.9Gi
```

**tldr:**
memory is fine. memory is not occupied and will be freed as soon as needed.

<details><summary>node exporter screenshot</summary>

<img width="1452" height="760" alt="Image" src="https://github.com/user-attachments/assets/7ba83ee3-5c83-4603-89e8-64baad323188" />

</details>

**Comment by 1NoOneKnowsMe1** [comment]
I’m experiencing the same issue. It only occurs during backups, where the memory usage increases  Unfortunately, the memory is not released after the process is complete.

<img width="613" height="332" alt="Image" src="https://github.com/user-attachments/assets/9e4a9b43-86c5-471b-a970-9fd9af2b2633" />

<img width="1602" height="385" alt="Image" src="https://github.com/user-attachments/assets/ceec42f1-9ba3-407a-b99c-27fc68c408b2" />
